earthmover/nodes/destination.py

```diff
@@ -132,3 +136,79 @@ def render_row(self, row: pd.Series):
             raise
 
         return json_string
+
+
+class CsvDestination(Destination):
+    """
+
+    """
+    mode: str = 'csv'  # Documents which class was chosen.
+    allowed_configs: Tuple[str] = (
+        'debug', 'expect', 'show_progress', 'repartition', 'source',
+        'extension', 'header', 'separator', 'limit', 'keep_columns'
+    )
+
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.header = self.error_handler.assert_get_key(self.config, 'header', dtype=bool, required=False, default=True)
+        self.separator = self.error_handler.assert_get_key(self.config, 'separator', dtype=str, required=False, default=",")
+        self.limit = self.error_handler.assert_get_key(self.config, 'limit', dtype=int, required=False, default=None)
+        self.extension = self.error_handler.assert_get_key(self.config, 'extension', dtype=str, required=False, default="csv")
+        self.keep_columns = self.error_handler.assert_get_key(self.config, 'keep_columns', required=False, default=None)
+        
+        self.file = os.path.join(
+            self.earthmover.state_configs['output_dir'],
+            f"{self.name}.{self.extension}"
+        )   
+
+    def execute(self, **kwargs):
+        """
+        
+        :return:
+        """
+        super().execute(**kwargs)
+        
+        self.data = self.upstream_sources[self.source].data
+
+        # Apply limit to dataframe if specified.
+        if self.limit:
```

Comment:
I'm not sure if raising an error is the right choice here. If the user specifies more rows than exist in the dataframe, we should just return all rows.